Mw/geo location improvements

michaelwood opened the pull request.

Commit:
frontend: search_helpers: Add display_name to selected output

When something is filtered add the display name to the selected item so
that we can display it along side the value.

Commit:
dataload: Add "Undetermined" for geo areas which have no value

This allows us to aggregate this value in the filters.

Commit:
csv_layout: Add the new geo additional_data fields

Also rename the existing ones to "best" to match the filter name

Commit:
import_to_elasticsearch: Add GeoCode for beneficiary/recipients

Commit:
csv_layout: Add new *GeoCode fields

Commit:
views: Add dev feature to return the grant data as json

This is useful to debug the fields and filtering which produced
the grant results.

Add ?json=true to a grant page to see the JSON representation

Commit:
import_to_elasticsearch: For recipient location use recipientOrganizationPostcode

This adds coverage for the 3rd source type
recipientOrganizationPostcode for geolook ups.

Commit:
dataload: import_to_elasticsearch: Remove field interdependency

In some geo areas some fields don't exist so avoid bailing out of the
whole process and handle each field's existence individually.

This is especially relevant to beneficiaryLocation due to the location
being able to be a geo area which is very large and therefore won't have
a single local authority.

Commit:
import_to_elasticsearch: Add keyword fields and process for geo fields

This adds the beneficiaryLocation recipientOrg location and county for
best available.

Commit:
frontend: search_helpers Rename aggregates to param_name

This is more accurate name for this var

Commit:
frontend: templates: Filters rework filters

- Add new filters for geo types and type switcher
- De-duplicate the filter template by switching to the generic
template. This also implements the finder within the filter and tidies
up some of the JS.
- Re-order filters
- Add new data fields for conditional display of filter group
- Create new group "Amount"
- Add search within filter
- Make County closed by default
- Make selection "sticky" between page reloads" for geo type selection
- Tidy up some JS lint issues
- Make currency closed by default

Commit:
frontend: views: Add the new aggregates/facets for geo types

Commit:
import_to_elasticsearch/views: Add new *County facets/aggregates

This is processing for best/recipient org/beneficiary location data for county
Adds an aggregated value for each

R2ZER0 reviewed (dismissed):
Looks good :+1: Approved pending updated test data, once the tests pass.

Commit:
templates: Fix linting error

Commit:
import_to_elasticsearch: If locationLookup doesn't exist skip

Prior versions of the data may not contain this field. Avoid bailing out
completely on this requirement. Especially useful for historical test
data

Commit:
test_data: Add new geo fields to test data

Commit:
views: Add in placeholders for url backward compatibility

These are the buckets which an old style json query url gets converted
into.

Commit:
tests_functional: Strengthen tests for currency filter

This makes the individual filter options able to be referenced from the
tests via id rather than relying on the dom layout improving the
longevity of the tests.

Also tidy up commented import.

Commit:
frontend: tests_functional: Add new test for county filter

rhiaro commented on dataload/import_to_elasticsearch.py:
```diff
@@ -483,6 +520,106 @@ def to_band(value, bins, labels):
+
+    # Prior versions of additional_data may not have this field
+    # or if locationLookup failed entirely for this grant
+    if not grant["additional_data"].get("locationLookup"):
```
is `grant["additional_data"]` always guaranteed to be set and the correct type at this point, or does this need a try/except?

michaelwood replied:
Yeah that should be covered by the earlier check on grant import, if it doesn't exist it will create an empty `dict`

https://github.com/ThreeSixtyGiving/grantnav/blob/live/dataload/import_to_elasticsearch.py#L420

rhiaro commented on dataload/import_to_elasticsearch.py:
```diff
@@ -483,6 +520,106 @@ def to_band(value, bins, labels):
+                    except KeyError:
+                        pass
+
+        # recipientOrganisationLocation
```
```suggestion
        # recipientOrganizationLocation
```

just for consistency..

Commit:
import_to_elasticsearch: Fix typo for nm->cd for code

GNRecipientOrgDistrictName got duplicated here.

Commit:
import_to_elasticsearch: Update comment

Update comment for spelling consistency

Co-authored-by: Amy Guy <amy@rhiaro.co.uk>
